opensearch-project/OpenSearch-Dashboards
# Add Dnd for Canvas in Vis Builder

### Description

This changes aims at extending Drag and Drop functionality to the canvas in Vis Builder. This feature will improve the getting started experience of users and enable them to create visualizations in an easier way.

Visualizations shown are based on the type and the field i.e being dropped into the canvas.
- Dropping a Categorical Field creates a Visualization with Count as Y-Axis and the dropped field as X-Axis
- Dropping a Numerical Field creates a Visualization with Dropped Field as Y-Axis and timeStamp as X-Axis
- Dropping Count creates a Visualization with Count as Y-Axis

### Issues Resolved




https://github.com/opensearch-project/OpenSearch-Dashboards/issues/7218

## Screenshot




https://github.com/opensearch-project/OpenSearch-Dashboards/assets/38322563/b627335f-3d8d-4887-b4f4-52feee989c57



## Testing the changes



- Navigate to the Vis Builder Home Page (ensure that you have appropriate data present)
- Drag a field from Field panel onto the Canvas
- Verify the visualization 

## Changelog


- feat: Add Dnd to Vis Builder Canvas
-
### Check List
